src/NUnitFramework/framework/Constraints/DelayedConstraint.cs

```diff
@@ -324,6 +326,39 @@ public override ConstraintResult ApplyTo<TActual>(ref TActual actual)
             return new DelegatingConstraintResult(this, BaseConstraint.ApplyTo(actual));
         }
 
+        /// <inheritdoc/>
+        public override async Task<ConstraintResult> ApplyToAsync<TActual>(Func<Task<TActual>> taskDel)
+        {
+            long now = Stopwatch.GetTimestamp();
+            long delayEnd = TimestampOffset(now, DelayInterval.AsTimeSpan);
+
+            if (PollingInterval.IsNotZero)
+            {
+                long nextPoll = TimestampOffset(now, PollingInterval.AsTimeSpan);
+                while ((now = Stopwatch.GetTimestamp()) < delayEnd)
+                {
+                    if (nextPoll > now)
+                        ThreadUtility.BlockingDelay((int)TimestampDiff(delayEnd < nextPoll ? delayEnd : nextPoll, now).TotalMilliseconds);
```

Comment:
Should we not use non-blocking' delay in an `async` method?
```suggestion
                        await Task.Delay(TimestampDiff(delayEnd < nextPoll ? delayEnd : nextPoll, now));
```

Reply:
Makes sense, done.